### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by Midnoclose** [binji/pokegb#7]
when building I get this error:
```
/usr/bin/ld: /tmp/ccloiK2Y.o: in function `main':
pokegb.cc:(.text.startup+0xe3): undefined reference to `SDL_Init'
/usr/bin/ld: pokegb.cc:(.text.startup+0x104): undefined reference to `SDL_CreateWindow'
/usr/bin/ld: pokegb.cc:(.text.startup+0x116): undefined reference to `SDL_CreateRenderer'
/usr/bin/ld: pokegb.cc:(.text.startup+0x138): undefined reference to `SDL_CreateTexture'
/usr/bin/ld: pokegb.cc:(.text.startup+0x144): undefined reference to `SDL_GetKeyboardState'
/usr/bin/ld: pokegb.cc:(.text.startup+0x2a4): undefined reference to `SDL_LockTexture'
/usr/bin/ld: pokegb.cc:(.text.startup+0x32c): undefined reference to `SDL_UnlockTexture'
/usr/bin/ld: pokegb.cc:(.text.startup+0x340): undefined reference to `SDL_RenderCopy'
/usr/bin/ld: pokegb.cc:(.text.startup+0x348): undefined reference to `SDL_RenderPresent'
/usr/bin/ld: pokegb.cc:(.text.startup+0x362): undefined reference to `SDL_PollEvent'
collect2: error: ld returned 1 exit status
```
I have libsdl2-dev installed, I'm not sure why it's not working

**Comment by LF-Lin** [comment]
The error may be the order of linking to static libraries, 

Replacing `$(CC) -O2 -Wall -Wno-return-type -Wno-misleading-indentation -o -lSDL2 $@ $<` 
by `$(CC) -O2 -Wall -Wno-return-type -Wno-misleading-indentation -o $@ $< -lSDL2 ` 
in Makefile.

It works for me (Linux ubuntu).

**Comment by M90K7** [comment]
build error
```
cc -O2 -Wall -Wno-return-type -Wno-misleading-indentation -lSDL2 -o pokegb pokegb.cc
pokegb.cc:1:10: fatal error: SDL2/SDL.h: No such file or directory
    1 | #include <SDL2/SDL.h>                                                                                      /******************************************/
      |          ^~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:2: pokegb] Error 1
```

**Comment by bioman3020** [comment]
I used windows 10 and codeblocks 20.03. I open the file pokegb and selected codeblocks as the IDE. I click build this error appears. 

![1](https://user-images.githubusercontent.com/72220146/120533368-a8e15e80-c3ae-11eb-8649-09564e5d1ef1.JPG)
![2](https://user-images.githubusercontent.com/72220146/120533383-abdc4f00-c3ae-11eb-96d3-40be573764cc.JPG)
![3](https://user-images.githubusercontent.com/72220146/120533397-ada61280-c3ae-11eb-8d77-b27c5ea62a5a.JPG)

**Comment by binji** [comment]
@M90K7 do you have SDL2 installed on your system? I'm assuming SDL2 is in your default include path, but it might not be.

@bioman3020 unforunately it won't compile on Windows directly. You can probably get it working w/ Windows Subsystem for Linux (WSL) though. I'm not sure, since I don't really develop much on Windows.

**Comment by Midnoclose** [comment]
> The error may be the order of linking to static libraries,
> 
> Replacing `$(CC) -O2 -Wall -Wno-return-type -Wno-misleading-indentation -o -lSDL2 $@ $<`
> by `$(CC) -O2 -Wall -Wno-return-type -Wno-misleading-indentation -o $@ $< -lSDL2 `
> in Makefile.
> 
> It works for me (Linux ubuntu).

This worked, thanks!

**Comment by Screwtapello** [comment]
I had the same problem as the OP, and moving `-lSDL2` to the end of the build command fixed it for me too (Debian Testing, gcc 10.2.1).

**Comment by binji** [comment]
Thanks, fixed here: https://github.com/binji/pokegb/commit/8bcba67ef4fb6f1e29928ab97249dc7f3f8eec20
